storage_service/locations/api/resources.py

```diff
@@ -1082,6 +1085,99 @@ def reingest_request(self, request, bundle, **kwargs):
 
         return self.create_response(request, response, status=status_code)
 
+    @_custom_endpoint(expected_methods=['post'])
+    def move_request(self, request, bundle, **kwargs):
+        r"""Request to move a stored AIP.
+
+        To test the move request, use the following sample command.
+        curl -d 'location_uuid=688c6909-9b97-43b0-9a32-4c6247bc0e84' \
+         -s -u test:test \
+        '/api/v2/file/3e8387db-2614-4215-8a19-f4b121ba1777/move/'
+        """
+        file = bundle.obj
+
+        if file.status != Package.UPLOADED:
+            response = {
+                'error': True,
+                'message': _('The file must be in an %s state to be moved.\
+                Current state: %s' % (Package.UPLOADED, file.status)),
+            }
+            return self.create_response(request,
+                                        response,
+                                        response_class=http.HttpBadRequest)
+
+        location_uuid = request.POST.get('location_uuid')
+
+        if not location_uuid:
+            return http.HttpBadRequest(_('All of these fields must be provided:\
+                                          location_uuid'))
+
+        try:
+            location = Location.objects.get(uuid=location_uuid)
+        except (Location.DoesNotExist, Location.MultipleObjectsReturned):
+            response = {
+                'error': True,
+                'message': _('Location UUID %(uuid)s \
+                failed to return a location') %
+                {'uuid': location_uuid},
+            }
+            return self.create_response(request,
+                                        response,
+                                        response_class=http.HttpBadRequest)
+
+        if location == file.current_location:
+            response = {
+                'error': True,
+                'message': _('New location must be different \
+                            to the current location'),
+            }
+            return self.create_response(request,
+                                        response,
+                                        response_class=http.HttpBadRequest)
+
+        if location.purpose != file.current_location.purpose:
+            response = {
+                'error': True,
+                'message': _('New location must have the same purpose \
```

Comment:
@sevein Perhaps I should modify this line in a different way so that it is both pep8-compliant but also doesn't create a space for the end user. I didn't google how to do that :stuck_out_tongue_closed_eyes: